### PR TITLE
feat: add angle and colour fields to block factory

### LINKED ISSUE #2290: Block factory: import plugin fields

### PR DESCRIPTION
**Pull request by maribethb**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

### Resolves

Fixes google/blockly#2159 

### Proposed Changes

- Adds colour and angle blocks to field blocks
- Uses plugin angle field for the colour block used to color the block factory block
- Does not add the multiline text since that's not in the original block factory either

### CODE REVIEW

File: examples/developer-tools/src/blocks/index.ts [hunk 1]
```diff
@@ -40,12 +45,18 @@ import '../output-generators/fields/label_serializable';
...
 import '../output-generators/fields/variable';
+import '../output-generators/fields/angle';
+import '../output-generators/fields/colour';
 import '../output-generators/colour';
 
 /* eslint-disable @typescript-eslint/naming-convention
  -- Blockly convention is to use snake_case for block names
 */
 export const registerAllBlocks = function () {
+  // Register the plugin fields before using them in blocks.
```
Comment (BeksOmega) [inline examples/developer-tools/src/blocks/index.ts:56]: Do they still need to be registered if we're constructing them explicitly?

File: examples/developer-tools/src/output-generators/fields/angle.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  generator.addHeaderLine(`registerFieldAngle();`);
+  return '';
+};
+
+scriptHeaderGenerator.forBlock['field_angle'] = function (
+  block: Blockly.Block,
+  generator: CodeHeaderGenerator,
+): string {
+  generator.addHeaderLine(
+    `<script src="https://unpkg.com/@blockly/field-angle"></script>`,
```
Comment (BeksOmega) [inline examples/developer-tools/src/output-generators/fields/angle.ts:63]: When we're importing from script tags does the field get registered automatically?

**Review by BeksOmega — APPROVED** [review]
Just so we don't lose it, we do have an open feature request for adding the mutiline field: https://github.com/google/blockly/issues/7206

File: examples/developer-tools/src/blocks/index.ts (hunk 1 above)
Comment (maribethb) [reply examples/developer-tools/src/blocks/index.ts:56]: Yeah, they need to be registered to use them in json, and the block preview is generated from the json block definition.

File: examples/developer-tools/src/output-generators/fields/angle.ts (hunk 2 above)
Comment (maribethb) [reply examples/developer-tools/src/output-generators/fields/angle.ts:63]: No, thanks, fixed

**Comment by maribethb** [comment]
> Just so we don't lose it, we do have an open feature request for adding the mutiline field: [google/blockly-samples#2326](https://github.com/google/blockly-samples/issues/2326)

Thanks! I'm planning to open follow up bugs to add all the plugin fields eventually. I'll move this issue to samples

**Pull request merged**